Transformations: Deprecate FormatTime transformation

Develer opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**

Deprecates and removes the FormatTime transformation. Format time is now redundant with convertFieldType and has been fully replaced.

What was done:
- Removed FormatTime transformation implementation
- Moved TimeFormat tests to convertFieldType
- Added migration to handle existing usage

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #105577 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

**Sam's Edit**

Ihor told me the Readme for how to write/run these kinds of migrations can now be found here:
[apps/dashboard/pkg/migration/README.md](https://github.com/grafana/grafana/blob/main/apps/dashboard/pkg/migration/README.md)

Develer committed:
chore(tests): move timeformat tests

Develer committed:
chore(migration): add migration

Develer committed:
chore(migration): remove formattime transformation

gelicia committed:
Make transformation docs

ivanortegaalba commented:
TODO:

- [ ] Implement the migration in the DashboardMigration.ts
- [ ] Add input with the different scenarios for this migration in the backend

ivanortegaalba reviewed (commented):
Make sure you use `grafanaBootData.settings.featureToggles.kubernetesDashboards` to enable the usage of App Platform to fetch any dashboard by default

ivanortegaalba reviewed (commented):
After a peer review, we decided:

1. Ivan will write a README about how to write a schemaVersion migration 
2. Once we have the documentation, we will update this PR with the following:
    1. Implement the migration in the [DashboardMigration.ts](https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/state/DashboardMigrator.ts)
    2. Add [input](https://github.com/grafana/grafana/tree/main/apps/dashboard/pkg/migration/testdata/input) with the different scenarios for this migration in the backend

Develer committed:
Merge branch 'main' into 105577-deprecate-formattime-transformation

Develer committed:
chore(migration): add frontend migration

Develer committed:
merge conflicts

Develer committed:
fix lint errors

Develer committed:
fix translation issues

ivanortegaalba commented on public/app/features/dashboard/state/DashboardMigrator.ts:
```diff
@@ -223,6 +225,7 @@ export class DashboardMigrator {
     // schema version 10 changes
     if (oldVersion < 10 && finalTargetVersion >= 10) {
       // move aliasYAxis changes
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Why do you `// eslint-disable-next-line @typescript-eslint/no-explicit-any`? Is there any new linting rule that breaks this?

ivanortegaalba commented on public/app/features/dashboard/state/DashboardMigrator.ts:
We should add tests in DashboardMigrator.test.ts to cover these changes too, WDYT? 😄 

- Panel with formatTime transformation → converts to convertFieldType
- Panel with full formatTime options (timeField, outputFormat, timezone)
- Panel with minimal formatTime options (only timeField)
- Panel with empty formatTime options
- Panel with multiple formatTime transformations
- Panel with mixed transformations (other transformers + formatTime)
- Nested panels in rows with formatTime
- Panel with no transformations (should remain unchanged)

ivanortegaalba reviewed (commented):
Code-wise looks good @Develer! Please have a look to the recommendations and let's try to get it merged!

Develer committed:
Merge branch 'main' into 105577-deprecate-formattime-transformation

Develer committed:
chore(linter): rollback suppressed errors

Develer committed:
add tests

Develer committed:
add backend output

Develer committed:
fix tests

ivanortegaalba committed:
Remove v42 latest files

ivanortegaalba committed:
Restore ID and remove extra output file

ivanortegaalba committed:
Merge remote-tracking branch 'origin/main' into 105577-deprecate-formattime-transformation

ivanortegaalba committed:
Add missing output files

ivanortegaalba committed:
Generate conversion files too

ivanortegaalba committed:
Regenerate v1 to v2 output files

ivanortegaalba reviewed (approved):
Reviewed and all the backend and frontend migrations were performed properly, and the implementation follows the standards, so this should be ready to be merged!

mckn reviewed (approved):
Looks good to me! I did a search for usages of the formatTime transformation and can't find any external code using it. So it should be safe to remove.

Do we have any documentation for users that might look for the formatTime transformation? Something telling them about what to use instead.

Develer committed:
fix merge conflicts

Develer commented:
@mckn @samjewell I've addressed some points [here](https://github.com/grafana/grafana/issues/105577#issuecomment-3547689283). Thank you for bringing it here.
